public/pages/utils/helpers.ts

```diff
@@ -96,8 +96,12 @@ export const filterAndSortDetectors = (
 export const getDetectorsToDisplay = (
   detectors: DetectorListItem[],
   page: number,
-  size: number
+  size: number,
+  dataSourceId: string
 ) => {
+  detectors.forEach((detector) => {
+    detector.dataSourceId = dataSourceId;
+  });
```

Comment:
If we are getting detectors to display, why are we mutating their `dataSourceId`s?

Reply:
I'm not mutating their dataSourceId here, just setting the new dataSourceId field in Detector interface so they can be passed to the next page